src/main.rs

```diff
@@ -110,7 +114,15 @@ async fn run_server() -> (
 
     common::create_frontend_config_file().expect("failed to create frontend config file");
 
-    let api = Api::new().await.expect("failed to create api");
+    type UiApi = Api<OmnectDeviceServiceClient, KeycloakProvider>;
+
+    let device_service = OmnectDeviceServiceClient::new(true)
+        .await
+        .expect("failed to create client to device service");
+    let keycloak_provider = KeycloakProvider::default();
+    let api = UiApi::new(device_service, keycloak_provider)
+        .await
+        .expect("failed to create api");
```

Comment:
I know we don't have such coding guidline, but I would shorten to:
```
let api = UiApi::new(
      OmnectDeviceServiceClient::new(true)
        .await
        .expect("failed to create client to device service"), 
      KeycloakProvider::default()
    )
    .await
    .expect("failed to create api");  
```

Decision is up to you (-;